### PR TITLE
Malicious site protection edge cases

### PR DESCRIPTION
**Pull request by kdzwinel**
I bundled couple of small changes:
- renamed "phishing detection" to "malicious site protection" to cover both phishing and malware and follow latest naming of this feature
- reorganized order of malicious site protection test pages on the index page for easier reading
- added three edge-case's: when site is in the local cache and doesn't require API request, when site is flagged by both DNS protection and malicious site protection, when site is excluded from protection via remote config exception

### CODE REVIEW

File: security/badware/index.html [hunk 1]
```diff
@@ -3,27 +3,30 @@
 
 <head>
```
Comment (kdzwinel) [inline security/badware/index.html:4]: The page looks like this after changes:

render

File: security/badware/index.html [hunk 2]
```diff
@@ -34,6 +37,18 @@ <h1>Phishing Detection Test Pages</h1>
...
+    <h2>Edge Cases</h2>
+    <ul>
+        <li><a href="http://malware.privacy-test-pages.site/security/badware/malware.html">Page flagged by both DNS protection and in-browser protection</a> (⚠️ different domain)</li>
+        <li><a href="http://broken.privacy-test-pages.site/security/badware/malware.html">Page where malicious site protection is disabled via remote config exception</a> (⚠️ different domain)</li>
+        <li><a href="./local-check.html">Page flagged in the local data (doesn't require API request)</a></li>
+    </ul>
+
+    <h2>Utility</h2>
+    <ul>
+        <li><a href="./phishing-iframe-top-navigator.html">Phishing iFrame Top Navigator</a></li>
```
Comment (kdzwinel) [inline security/badware/index.html:50]: BTW, do we even have to link to it? I think it only confuses folks since it's not a test.

File: security/badware/index.html [hunk 3]
```diff
@@ -3,27 +3,30 @@
...
-        <li><a href="./phishing-meta-redirect.html">Phishing Redirect via Meta Refresh (Flagged in Dataset)</a></li>
         <li><a href="./phishing-popups.html">Phishing Open via Popups</a></li>
         <li><a href="./phishing-url-tampering.html">Phishing Opening with URL Tampering</a></li>
         <li><a href="./phishing-form-submission.html">Phishing Form Submission</a></li>
-        <li><a href="./phishing-iframe-top-navigator.html">Phishing iFrame Top Navigator</a></li>
         <li><a href="./phishing-service-worker.html">Phishing Service Worker</a></li>
+        <li><a href="./malware-download.html">Malware Download Test</a></li>
+    </ul>
+
+    <h2>Redirects</h2>
```
Comment (kdzwinel) [inline security/badware/index.html:24]: Not sure if it's a useful grouping, just wanted to make things more findable given we now have a ton of those pages. I'm happy to revert this change if it doesn't make sense

Comment (not-a-rootkit) [reply security/badware/index.html:24]: No this makes perfect sense. Thanks Konrad!

File: security/badware/index.html (hunk 2 above)
Comment (not-a-rootkit) [reply security/badware/index.html:50]: We can remove this link actually, it's not useful on its own.

**Review by not-a-rootkit — APPROVED** [review]
Happy with this, appreciate it @kdzwinel !

File: security/badware/index.html (hunk 2 above)
Comment (kdzwinel) [reply security/badware/index.html:50]: Awesome, I dropped it
